core/chains/evm/client/simulated_backend_client.go

```diff
@@ -388,21 +388,62 @@ func (c *SimulatedBackendClient) SuggestGasPrice(ctx context.Context) (*big.Int,
 
 // BatchCallContext makes a batch rpc call.
 func (c *SimulatedBackendClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) error {
+	select {
+	case <-ctx.Done():
+		return errors.New("context canceled")
+	default:
+		//do nothing
+	}
```

Comment:
To simulate the way this RPC call handles contexts correctly, would we need to do this check every time in the loop or is once enough? I guess it doesn't matter that much, though I'm curious how closely we try to simulate the backends.

Reply:
I think once should be ok because the loop will run really quickly so it will be anyways hard to test context timeout 
behavior. It doesn't seem like underlying `SimulatedBackend` is meant for mocking either, so no good way to introduce artificial delay.